ClientApp/src/components/Chat.js

```diff
@@ -0,0 +1,94 @@
+import React, { useEffect, useState, useContext } from 'react';
+import { HubConnectionBuilder } from '@microsoft/signalr';
+import { UserContext } from '../UserContext';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faPaperPlane } from '@fortawesome/free-solid-svg-icons';
+import '../chat.css';
+
+//TODO: RegEx expression for messages, save messages and load up on render.
+
+export const Chat = (props) => {
+    const [message, setMessage] = useState([]);
+    const [messages, setMessages] = useState([]); //array of objects with message.type, message.nick, message.text
+    const [hubConnection, setHubConnection] = useState();
+    const { user, loggedIn } = useContext(UserContext);
+
+    useEffect(() => {
+        if (loggedIn) {
+            const connection = new HubConnectionBuilder()
+                .withUrl("/ChatHub")
+                .build();
+
+            connection.on("ReceiveGroupMessage", (nick, receivedMessage) => receiveMessage(nick, receivedMessage));
+
+            connection.logging = true;
+            connection
+                .start()
+                .then(() => console.log('Connection established.'))
+                .catch(err => console.log(`Error while establishing connection: ${err}`))
+                .then(() => connection.invoke("AddToGroup", props.groupId))
+                .catch(err => console.error(`Error while adding to group: ${err}`));
+
+            setHubConnection(connection);
+        }
+    }, [loggedIn]);
+
+    const sendMessage = async () => {
+        if (message !== '') {
+            await hubConnection
+                .invoke('SendGroupMessage', props.groupId, user.firstName, message)
+                .catch(err => console.error(`Error while sending message: ${err}`));
+
+            setMessage('');
+        }
+    }
+
+    const receiveMessage = (nick, receivedMessage) => {
+        var tempMessage = new Object();
+        tempMessage.nickname = nick;
+        tempMessage.text = receivedMessage;
+        tempMessage.type = "in";
+        if (user.firstName === nick)
+            tempMessage.type = "out";
+        
+        messages.push(tempMessage);
+        var array = new Array();
+        setMessages(messages.concat(array));
```

Comment:
I don't really understand why are you concatenating an empty array here?
Simpler: setMessages(messages => messages.concat(tempMessage));

Reply:
It used to not update the component and re-render it without concatenating an empty list, I see your suggested solution works too, so I've imported it.